### ISSUE TITLE
aws adapter get_path should work without CF distribution or NGINX proxy

### ISSUE THREAD

**Issue by camallen** [zooniverse/panoptes#2308]
The aws storage adapter uses a domain prefix that is meant to  resolves to an S3 bucket path. In our production / staging setup we use a CloudFront distribution to proxy the prefix CNAME to the s3 bucket path URL. 

An out of the box installation shouldn't require another service (nginx / CF) to resolve the `get_path` to the correct s3 bucket URL

e.g. storage.yml 
``` yaml
development:
  adapter: aws
  prefix: development
  bucket: labelmars-image-data
  access_key_id: <key>
  secret_access_key: <secret>
```

sets an avatar media resource src as `development/subject_location/12d7141b25-0353-4a7e-aee8-480c351521d3.jpeg` and thus the aws adapter [get_path](https://github.com/zooniverse/Panoptes/blob/20483e32c5547300c4e7c2bb630f6c18e6b9f356/app/services/media_storage/aws_adapter.rb#L40) code returns 
`https://development/subject_location/12d7141b25-0353-4a7e-aee8-480c351521d3.jpeg`

We should provide a configuration directive to modify the default behaviour to not use a CDN or maybe find the bucket URL programatically or even allow the dev to add a `domain_prefix` yaml option to ensure get_paths return the s3 bucket URL's. 
``` yaml
  domain_prefix: labelmars-image-data.s3.amazonaws.com
```
that results in `https://labelmars-image-data.s3.amazonaws.com/development/subject_location/12d7141b25-0353-4a7e-aee8-480c351521d3.jpeg`

Thanks to @squiddon for bringing this up.

**Comment by astopy** [comment]
See also #1763

**Comment by camallen** [comment]
keeping this open

**Comment by lcjohnso** [comment]
closed by #3453
